aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -1024,7 +1025,31 @@ internal class RealAWSCognitoAuthPlugin(
         onSuccess: Action,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
+            when (authState.authNState) {
+                // Check if user signed in
+                is AuthenticationState.SignedIn -> {
+                    GlobalScope.launch {
+                        try {
+                            val accessToken = getSession().userPoolTokens.value?.accessToken
+                            val verifyUserAttributeRequest = VerifyUserAttributeRequest.invoke {
+                                this.accessToken = accessToken
+                                this.attributeName = attributeKey.keyString
+                                this.code = confirmationCode
+                            }
+                            authEnvironment.cognitoAuthService
+                                .cognitoIdentityProviderClient?.verifyUserAttribute(
+                                    verifyUserAttributeRequest
+                                )
+                            onSuccess.call()
```

Comment:
Isn't `verifyUserAttributeRequest` a suspended function? 
I am afraid `onSuccess.call()` can run concurrently with `verifyUser..` and return false positive. You should chain the `onSuccess` with `verifyUser..`.